### PR TITLE
Make operator install KIC CRDs from manifests

### PR DESCRIPTION
**Pull request by Rulox**
### Proposed changes
We used to create the CRDs from the Operator, the only difference is that now, we bundle the full manifests instead of creating them from go code.

This will make sure all our KIC CRDs are working for the given version of the operator-kic relationship, and will also have the openapi schema validation that was not present before.

Downsides
* We don't know if this is the preferred method as OLM/operator-sdk team haven't been able to provide us solutions either on Slack or GH issues, and our operator is a RH certified operator, so it has certain restrictions with the CSV.

### CODE REVIEW

File: pkg/controller/nginxingresscontroller/crds.go [hunk 1]
```diff
@@ -1,106 +1,59 @@
...
+	var crds []*v1beta1.CustomResourceDefinition
+	for _, path := range manifests {
+		f, err := os.Open(path)
+		if err != nil {
+			return nil, fmt.Errorf("failed to open the CRD manifest %v: %v", path, err)
+		}
+
+		var crd *v1beta1.CustomResourceDefinition
+
+		err = yaml.NewYAMLOrJSONDecoder(f, decoderBufferSize).Decode(&crd)
```
Comment (pleshakov) [inline pkg/controller/nginxingresscontroller/crds.go:44]: hm. we're passing a pointer to a pointer here.  I haven't run it, but just double checking, does it work as expected?

File: pkg/controller/nginxingresscontroller/nginxingresscontroller_controller.go [hunk 2]
```diff
@@ -49,6 +51,66 @@ func newReconciler(mgr manager.Manager, sccAPIExists bool) reconcile.Reconciler
...
+	reqLogger := log.WithValues("Request.Namespace", "", "Request.Name", "nginxingresscontroller-controller")
+
+	// Create CRDs with a different client (apiextensions)
+	apixClient, err := apixv1beta1client.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		reqLogger.Error(err, "unable to create client for CRD registration")
+		return err
+	}
+
+	if isLocal() {
```
Comment (pleshakov) [inline pkg/controller/nginxingresscontroller/nginxingresscontroller_controller.go:76]: shall we check it before we create a client?

File: README.md [hunk 3]
```diff
@@ -27,9 +27,13 @@ It is possible to run the operator in your local machine. This is useful for tes
     ```
     kubectl apply -f deploy/crds/k8s.nginx.org_nginxingresscontrollers_crd.yaml
     ```
+1. Apply the NGINX Ingress Controller CRDs:
```
[the remark this answers is not in the document]
Comment (Rulox) [reply README.md:30]: No, actually I need to remove this from the README. The operator applies the manifests automatically even when running in local.

File: pkg/controller/nginxingresscontroller/crds.go (hunk 1 above)
Comment (Rulox) [reply pkg/controller/nginxingresscontroller/crds.go:44]: Hmm yeah it works 🤔 , I'll remove the `*` and test, it shouldn't be a pointer, thanks

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Rulox) [reply README.md:30]: This is only needed when running the operator locally (for development for instance)

File: README.md [hunk 4]
```diff
@@ -27,9 +27,13 @@ It is possible to run the operator in your local machine. This is useful for tes
...
+    ```
+    kubectl apply -f deploy/kic_crds
```
Comment (Dean-Coakley) [inline README.md:32]: https://github.com/nginxinc/nginx-ingress-operator/tree/add-kic-crds/deploy/kic_crds/ does not exist?

I think this was intended to be:
https://github.com/nginxinc/nginx-ingress-operator/tree/add-kic-crds/build/kic_crds/

```suggestion
    kubectl apply -f build/kic_crds
```

Comment (Rulox) [reply README.md:32]: Yess good catch @Dean-Coakley thanks

**Pull request merged**